airbyte-integrations/connectors/source-shopify/unit_tests/test_transform.py

```diff
@@ -0,0 +1,140 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from decimal import Decimal
+
+import pytest
+from source_shopify.transform import Transformer
+
+
+def find_by_path(path_list, value):
+    key_or_index = path_list.pop(0)
+    result_value = value[key_or_index]
+    if len(path_list) > 0:
+        result_value = find_by_path(path_list, result_value)
+    return result_value
+
+
+@pytest.mark.parametrize(
+    "transform_object, schema, checks",
+    [
+        (
+            {
+                "id": 1,
+                "email": "airbyte-integration-test@airbyte.io",
+                "buyer_accepts_marketing": False,
+                "created_at": "2021-07-08T04:58:50-07:00",
+                "note": None,
+                "total_weight": 112,
+                "shipping_lines": [{"price": "20.17"}],
+                "line_items": [{"gift_card": False, "grams": 112, "line_price": "19.00", "price": "19.00"}],
+                "total_discounts": "0.00",
+                "total_line_items_price": "19.00",
+                "total_price": "39.17",
+                "total_tax": "0.00",
+                "subtotal_price": "19.00",
+                "customer": {"total_spent": "0.00"},
+            },
+            {
```

Comment:
could you isolate the test inputs as much as possible so that it is clear what each input is testing? 

For example, the current input tests many things: 
1. A field with the correct type is not transformed
2. A string field which should be a number is transformed
3. A field which is nested in an object is transformed
4. A field which is nested in an object inside an array is transformed

Could we then have at least 4 test cases, each of which specifically test those? 

The guiding principle here is that tests should be as isolated as possible so that: 
1. a failure clearly indicates what is wrong
2. the reader can have confidence that all tests cases are tested (right now the monolithic test case makes it very difficult to know all test cases are accurately tested)

I think we also need an additional test case that `array` which contains strings but that should be ints should also be tested right? Currently all items inside arrays are objects in this example